pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -649,7 +655,9 @@ public void shutdown() throws PulsarClientException {
         try {
             lookup.close();
             cnxPool.close();
-            timer.stop();
+            if (needStopTimer) {
+                timer.stop();
+            }
```

Comment:
Could you please add a test to cover this branch?

Reply:
ok, done